### PR TITLE
Ad hk migrate 60 712

### PR DESCRIPTION
**Pull request by JohannaOtt**
SVC-1463

### CODE REVIEW

**Review by florianesser — CHANGES_REQUESTED** [review]
## Test data

Versions should generally not be abbreviated by leaving out the dots, e.g. "5.0" should never become "50". In this PR, a new directory "testdaten/Hauskoordinaten/50" is added which should be called "testdaten/Hauskoordinaten/5.0" instead.

Also, when a subdirectory for version 5.0 is introduced, the 4.0 schema should go into its own directory ("testdaten/Hauskoordinaten/4.0").

## Comments on commits and messages

### `feat: add HK model documentation`

- Message should be `docs: add HK model documentation`

### `fixup: add documentation/test data`

This commit contains
- a fixup that should be squashed into the previous commit
- the addition of a new test file that should have its own commit (e.g. `test: add test data for HK model v5.0`)

### `feat: add alignment with HK 5.0 schema`

> that were not find anymore

that were not found anymore

File: annex-1/mappings7.1/Addresses/hauskoordinaten-ad-01.halex.alignment.xml [hunk 1]
```diff
@@ -0,0 +1,1538 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<alignment xmlns="http://www.esdi-humboldt.eu/hale/alignment">
+    <base prefix="ba1" location="../../mappings6.0/base-functions.halex.alignment.xml"/>
+    <customFunction>
+        <cf:custom-property-function xmlns:cf="http://www.esdi-humboldt.eu/hale/custom-function" identifier="adv.inspire.geometry.normalizedPoint" name="AdV Punkt aus Ordinaten erstellen" type="groovy">
+            <cf:input eager="false" maxOccurs="1" minOccurs="1" name="northing">
+                <cf:binding>java.lang.String</cf:binding>
```
Comment (florianesser) [inline annex-1/mappings7.1/Addresses/hauskoordinaten-ad-01.halex.alignment.xml:7]: I think the binding needs to be `java.lang.Double`. With `java.lang.String` I get the following transformation error when trying to transform the test dataset:

```
Caused by: groovy.lang.MissingMethodException: No signature of method: java.lang.String.div() is applicable for argument types: (BigDecimal) values: [1000000.0]
Possible solutions: is(java.lang.Object), wait(), trim(), size(), size(), md5()
	at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.unwrap(ScriptBytecodeAdapter.java:70)
	at org.codehaus.groovy.runtime.callsite.PojoMetaClassSite.call(PojoMetaClassSite.java:46)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:115)
	at org.kohsuke.groovy.sandbox.impl.Checker$1.call(Checker.java:120)
	at org.kohsuke.groovy.sandbox.GroovyInterceptor.onMethodCall(GroovyInterceptor.java:21)
	at eu.esdihumboldt.util.groovy.sandbox.internal.RestrictiveGroovyInterceptor.onMethodCall(RestrictiveGroovyInterceptor.java:305)
	at org.kohsuke.groovy.sandbox.impl.Checker$1.call(Checker.java:118)
	at org.kohsuke.groovy.sandbox.impl.Checker.checkedCall(Checker.java:115)
	at org.kohsuke.groovy.sandbox.impl.Checker.checkedBinaryOp(Checker.java:318)
	at org.codehaus.groovy.vmplugin.v7.IndyInterface.selectMethod(IndyInterface.java:234)
	at Script1.run(Script1.groovy:2)
	at eu.esdihumboldt.util.groovy.sandbox.DefaultGroovyService.evaluate(DefaultGroovyService.java:189)
	at eu.esdihumboldt.hale.ui.service.groovy.internal.PreferencesGroovyService.evaluate(PreferencesGroovyService.java:154)
	at eu.esdihumboldt.cst.functions.groovy.GroovyTransformation.evaluate(GroovyTransformation.java:127)
	at eu.esdihumboldt.hale.common.align.custom.groovy.CustomGroovyTransformation.evaluate(CustomGroovyTransformation.java:104)
	... 29 more
```

File: annex-1/mappings7.1/Addresses/hauskoordinaten-ad-01.halex.alignment.xml [hunk 2]
```diff
@@ -0,0 +1,1538 @@
...
+            </cf:input>
+            <cf:input eager="false" maxOccurs="1" minOccurs="1" name="easting">
+                <cf:binding>java.lang.String</cf:binding>
```
Comment (florianesser) [inline annex-1/mappings7.1/Addresses/hauskoordinaten-ad-01.halex.alignment.xml:10]: Here as well

File: annex-1/mappings7.1/Addresses/hauskoordinaten-ad-01.halex [hunk 3]
```diff
@@ -0,0 +1,1156 @@
...
+    </save-config>
+    <resource action-id="eu.esdihumboldt.hale.io.schema.read.source" provider-id="eu.esdihumboldt.hale.io.csv.reader.schema">
+        <setting name="charset">UTF-8</setting>
+        <setting name="resourceId">ecf01797-9cb4-4592-90d7-328216f6185b</setting>
+        <setting name="types">java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.lang.String</setting>
+        <setting name="quote">"</setting>
+        <setting name="skip">1</setting>
+        <setting name="source">file:/C:/Users/Johanna%20Ott/Box%20Sync/staff/Projects/2020-27%20AdV%20GeoInfoDok%207.1%20Migration/Data%20and%20documentation/Daten%20Hauskoordinaten/Testdaten_Hauskoordinaten_Sommer_2023/adressen-testdaten.txt</setting>
```
[the remark this answers is not in the document]
Comment (JohannaOtt) [reply annex-1/mappings7.1/Addresses/hauskoordinaten-ad-01.halex]: Done

File: annex-1/mappings7.1/Addresses/hauskoordinaten-ad-01.halex.alignment.xml (hunk 1 above)
Comment (JohannaOtt) [reply annex-1/mappings7.1/Addresses/hauskoordinaten-ad-01.halex.alignment.xml]: Done

File: annex-1/mappings7.1/Addresses/hauskoordinaten-ad-01.halex.alignment.xml (hunk 2 above)
Comment (JohannaOtt) [reply annex-1/mappings7.1/Addresses/hauskoordinaten-ad-01.halex.alignment.xml]: Done

**Pull request merged**